pkg/stanza/fileconsumer/config_test.go

```diff
@@ -352,6 +352,14 @@ func TestUnmarshal(t *testing.T) {
 					return newMockOperatorConfig(cfg)
 				}(),
 			},
+			{
+				Name: "poll_file_limit_1",
+				Expect: func() *mockOperatorConfig {
+					cfg := NewConfig()
+					cfg.PollFileLimit = 1
```

Comment:
We don't need to address this for this PR, but I was surprised this is valid, given that the default value for `MaxConcurrentFiles` is 1024. Really we ought to validating against the default config + modifications, rather than a from-scratch config that is not representative.

I'll make an issue to track this.